website/docs/language/tests/index.mdx

```diff
@@ -1009,4 +1010,143 @@ Run D (parallel: true)
+  }
+}
+```
+
+The `backend` block here is like the pre-existing `backend` block that can be included in the `terraform` block.
+Any of the current state-storage backends in Terraform can be used. Refer to the main [backend documentation](https://developer.hashicorp.com/terraform/language/backend)
+for a list of available backends and how to configure them.
+
+-> **Note:** Pay attention to any environment variables that you use to configure backends during `plan` and `apply`
+operations; these will also be used during tests. You may need to change your environment depending on which workflow
+you are performing.
+
+### Only one backend can be used per internal state
+
+A `backend` block is used to set the initial value of an internal, in-memory state used during test execution. Due to this,
+a `backend` block can only be used in the first run block that represents an `apply` operation for a given internal state.
+
+As described in the [Modules State](#modules-state) section, a run block's internal state is determined either by the module
+that's under test or by explicitly setting a `state_key` argument. The example test file below shows how separate `backend` blocks
+can be used in the same file as long as they are linked to different states. Those states may be implied or explicitly declared.
+
+
+```hcl
+# multiple-backends.tftest.hcl
+
+# Root module state
+## This backend block implicitly controls the internal state for testing the root module
+run "test-main-1" {
+  backend "local" {
+    path = "./state/test_A.tfstate"
+  }
+}
+
+run "test-main-2" {
+  # No backend block can be used here;
+  # this is the second apply run block for the root module's state
+}
+
+# Child module state
+## This backend block implicitly controls the internal state for the module "./modules/my-module"
+run "test-module-1" {
+  module {
+    source = "./modules/my-module"
+  }
+
+  backend "local" {
+    path = "./state/my-module.tfstate"
+  }
+}
+
+run "test-module-2" {
+  module {
+    source = "./modules/my-module"
+  }
+  # No backend block can be used here;
+  # this is the second apply run block for the "./modules/my-module" module's state
+}
+
+# State key state
+## This backend block controls the internal state for the explicitly-declared "different-state" state
+run "test-state-key-1" {
+  state_key = "different-state"
+
+  backend "local" {
+    path = "./state/state-key.tfstate"
+  }
+}
+
+run "test-state-key-2" {
+  state_key = "different-state"
+  # No backend block can be used here;
+  # this is the second apply run block for the state_key "different-state"
+}
+```
+
+In the example above there are two `run` blocks per internal state, and all `run` blocks represent an `apply` operation.
+Only the first `run` block in each pair can include the `backend` block. 
+
+-> **Note:** Terraform will stop you from trying to re-use the same state file for multiple internal states. This prevents
+multiple tests attempting to write to the same state.
```

Comment:
This line hints at the overlap of backends and skip_cleanup, which is a currently undocumented (in this PR) feature of backends in test

Reply:
I think we definitely would have a PR in the end that reviews the feature's docs as a whole

Reply:
I'll make a ticket for a final look over the docs before we have the PR into `main`, in case we need to connect sections that were added in different PRs